### PR TITLE
feat: allow adding routes without using decorator

### LINKED ISSUE #587: Add an option to add routes without decorator
In some cases, users may want to add routes without using the decorator.
In Flask it is supported with [add_url_rule](https://tedboy.github.io/flask/generated/generated/flask.Flask.add_url_rule.html)

We can use the same syntax:
```py
from robyn import Robyn

app = Robyn(__file__)

def hello(request):
    return "Hello, world!"

app.add_url_rule('/', 'hello', hello)

app.start(port=8080)
```

This can be useful in cases where you are initiating the app in a certain part of your program but would like to retroactively add routes.

### PR DESCRIPTION
**Pull request by IdoKendo**
**Description**

This PR fixes #587 by making add_route public, while adding tests to validate the behavior.

### CODE REVIEW

**Review by sansyrox — COMMENTED** [review]
Hey @IdoKendo 👋 

Thank you for the PR 😄 I have one question - don't views(https://sparckles.github.io/robyn/#/features?id=views) serve a similar purpose?

As we support functional views in Robyn

**Comment by IdoKendo** [comment]
> Hey @IdoKendo 👋
> 
> Thank you for the PR 😄 I have one question - don't views(https://sparckles.github.io/robyn/#/features?id=views) serve a similar purpose?
> 
> As we support functional views in Robyn

Hey @sansyrox - Yeah, it's similar to serving views using app.add_view - only for non-view handlers.
Since we allow functional views, why not for non-view handlers? 😃 
I've seen several use-cases that can benefit from it and it's an overall minor change ✌️

File: integration_tests/base_routes.py [hunk 1]
```diff
@@ -714,6 +723,24 @@ async def async_auth(request: Request):
...
+async def async_without_decorator():
+    return "Success!"
+
+
+app.add_route(HttpMethod.GET, "/sync/get/no_dec", sync_without_decorator)
+app.add_route(HttpMethod.PUT, "/sync/put/no_dec", sync_without_decorator)
+app.add_route(HttpMethod.POST, "/sync/post/no_dec", sync_without_decorator)
+app.add_route(HttpMethod.GET, "/async/get/no_dec", async_without_decorator)
+app.add_route(HttpMethod.PUT, "/async/put/no_dec", async_without_decorator)
+app.add_route(HttpMethod.POST, "/async/post/no_dec", async_without_decorator)
```
Comment (sansyrox) [inline integration_tests/base_routes.py:739]: Hey @IdoKendo 👋 

Thank you for the PR 😄 And apologies for the late review

I was wondering if we are exposing a public API for this, we should also allow strings as a method specifier.

e.g. `app.add_router("GET". "/sync/get")`

Comment (IdoKendo) [reply integration_tests/base_routes.py]: Yeah, sounds good, added a consideration for it!

**Review by sansyrox — APPROVED** [review]
Looks good to me 🔥 

Great work! Can we please add some documentation for this?

**Comment by sansyrox** [comment]
Thanks @IdoKendo 😄

**Review by sansyrox — APPROVED** [review]
LGTM! 😄

**Pull request merged**